### PR TITLE
Np 3052 unconfirmed journal and journal

### PR DESCRIPTION
**Pull request by brinxmat**
- Replaces old Journal with new Journal
- Adds UnconfirmedJournal

### CODE REVIEW

**Review by janbj — COMMENTED** [review]
Replaces old Journal with new Journal?

**Review by axthosarouris — DISMISSED** [review]

File: src/main/java/no/unit/nva/model/contexttypes/Journal.java [hunk 1]
```diff
@@ -1,96 +1,33 @@
...
-        public Builder() {
-        }
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+public class Journal implements Periodical, BasicContext {
+    private final URI id;
 
-        public Journal.Builder withTitle(String title) {
-            this.title = title;
-            return this;
+    public Journal(@JsonProperty("id") String id) {
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/contexttypes/Journal.java:18]: `@JsonCreator`

File: src/main/java/no/unit/nva/model/contexttypes/Periodical.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package no.unit.nva.model.contexttypes;
+
+/**
+ * Marker pattern parent interface for Journal and UnconfirmedJournal.
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/contexttypes/Periodical.java:4]: Although I understand the "parent interface", I do not understand the "pattern parent interface" and the "marker pattern parent interface"

File: src/test/java/no/unit/nva/model/contexttypes/ReportTest.java [hunk 3]
```diff
@@ -35,15 +35,15 @@
     @DisplayName("Reports can be created")
     @ParameterizedTest
     @CsvSource({
-        "Series,123,A publisher,9780201309515,0363-6941,1945-662X",
-        "Series,123,A publisher,9780201309515|9788131700075,0363-6941,1945-662X",
-        "Series,123,A publisher,9780201309515|9788131700075,0363-6941,1945-662X",
-        "Series,123,A publisher,9780201309515|9788131700075,0363-6941,1945-662X",
-        "Series,,A publisher,9780201309515|9788131700075,0363-6941,1945-662X",
+        "Journal,123,A publisher,9780201309515,0363-6941,1945-662X",
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/contexttypes/ReportTest.java:38]: I recommend changing the value to "Journal title" and changing the name of the respective parameter as well

File: src/test/java/no/unit/nva/model/ModelTest.java [hunk 4]
```diff
@@ -307,12 +307,11 @@ private static PublicationContext generateReportContext()
             .build();
     }
 
-    private static PublicationContext generateJournalContext() throws InvalidIssnException, MalformedURLException {
-        return new Journal.Builder()
-            .withOnlineIssn("1111-1119")
-            .withPrintIssn("2222-2227")
-            .withTitle("The journal of mechanically separated meats")
-            .build();
+    private static PublicationContext generateUnconfirmedSeries() throws InvalidIssnException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/ModelTest.java:310]: Change the method name to be in sync with terminology used in the rest of the model.

File: src/test/java/no/unit/nva/model/instancetypes/PeriodicalTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        String json = "{ \"type\": \"UnconfirmedJournal\", \"onlineIssn\": null }";
+
+        UnconfirmedJournal journal = objectMapper.readValue(json, UnconfirmedJournal.class);
+
+        assertNotNull(journal);
+        assertNull(journal.getPrintIssn());
+    }
+
+    @Test
+    public void seriesConstructorCreatesObjectWithId() throws JsonProcessingException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/instancetypes/PeriodicalTest.java:61]: change test name

File: src/test/java/no/unit/nva/model/instancetypes/PeriodicalTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        String json = "{ \"type\": \"Journal\", \"id\": \"https://example.org/series\" }";
+        Journal journal = objectMapper.readValue(json, Journal.class);
+        assertNotNull(journal);
+        assertNotNull(journal.getId());
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    void seriesConstructorThrowsExceptionWhenIdIsInvalid(String value) {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/model/instancetypes/PeriodicalTest.java:70]: chnage test name

**Review by axthosarouris — COMMENTED** [review]
Many types are now "UnconfirmedJournal", I suppose this is fixed in another PR?

**Review by janbj — APPROVED** [review]

File: src/test/java/no/unit/nva/model/contexttypes/ReportTest.java (hunk 3 above)
Comment (brinxmat) [reply src/test/java/no/unit/nva/model/contexttypes/ReportTest.java:38]: Fixed in following PR

File: src/test/java/no/unit/nva/model/ModelTest.java (hunk 4 above)
Comment (brinxmat) [reply src/test/java/no/unit/nva/model/ModelTest.java:310]: Removed in next PR

File: src/main/java/no/unit/nva/model/contexttypes/Journal.java (hunk 1 above)
Comment (brinxmat) [reply src/main/java/no/unit/nva/model/contexttypes/Journal.java:18]: Fixed in next PR

**Pull request merged**